ote_cli/ote_cli/utils/io.py

```diff
@@ -114,3 +141,83 @@ def generate_label_schema(dataset, task_type):
+
+
+def create_task_from_deployment(openvino_task_class, deployed_code_zip_path):
+    """
+    Creates a child class of passed 'openvino_task_class', instance of which is initialized by deployment (zip archive).
+    """
+
+    class Task(openvino_task_class):
+        """A child class of 'openvino_task_class', instance of which is initialized by deployment (zip archive)."""
+
+        class Inferencer:
+            """ModelAPI-based OpenVINO inferencer."""
+
+            def __init__(self, model, converter) -> None:
+                self.model = model
+                self.converter = converter
+
+            def predict(self, frame):
+                """Returns predictions made on a given frame."""
+
+                dict_data, input_meta = self.model.preprocess(frame)
+                raw_result = self.model.infer_sync(dict_data)
+                predictions = self.model.postprocess(raw_result, input_meta)
+                annotation_scene = self.converter.convert_to_annotation(
+                    predictions, input_meta
+                )
+                return annotation_scene
+
+        def __init__(self, task_environment) -> None:
+            self.task_environment = task_environment
+            with tempfile.TemporaryDirectory() as temp_dir:
+                with ZipFile(deployed_code_zip_path) as myzip:
+                    myzip.extractall(temp_dir)
+                with ZipFile(
+                    os.path.join(
+                        temp_dir, "python", "demo_package-0.0-py3-none-any.whl"
+                    )
+                ) as myzip:
+                    myzip.extractall(temp_dir)
+
+                model_path = Path(os.path.join(temp_dir, "model", "model.xml"))
+                config_path = Path(
+                    os.path.join(temp_dir, "demo_package", "config.json")
+                )
+
+                with open(config_path, encoding="UTF-8") as read_file:
+                    parameters = json.load(read_file)
+                converter_type = Domain[parameters["converter_type"]]
+
+                self.inferencer = self.Inferencer(
+                    create_model(model_path, config_path),
+                    create_converter(
+                        converter_type, self.task_environment.label_schema
+                    ),
+                )
+
+        def infer(
+            self,
+            dataset: DatasetEntity,
+            inference_parameters: Optional[InferenceParameters] = None,
+        ) -> DatasetEntity:
+            """Inference method."""
+            if inference_parameters is not None:
+                update_progress_callback = inference_parameters.update_progress
+            dataset_size = len(dataset)
+            for i, dataset_item in enumerate(dataset, 1):
+                predicted_scene = self.inferencer.predict(dataset_item.numpy)
+                if (
+                    self.task_environment.model_template.task_type
```

Comment:
Should model template be specified when working with deployment? If yes, why do we need to reimplement the task, rather than use the one from the template?

Reply:
Model template should be specified because of:
1. need of `evaluate` method which is taken from original task.
2. need of `task_type` for `infer` method details.

For sure deployed **model** testing via original task can be implemented without creating child class of the task. We will need to read deployed model stored in different format from format of openvino task. It is possible, but I would also add other parts of deployed **code** in the inference pipeline:
- self.model.preprocess()
- self.model.infer_sync()
- self.model.postprocess()
- self.converter.convert_to_annotation()

It allows testing probably not whole deployment, but almost everything except demo part.

Reply:
@druzhkov-paul  please have a look one more time